### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by fuzzball81**
Adding an initial directory structure as well as support files for:

- CI: .travis.yml
- Auto pull request merge: .mergify.yml
- Pipenv: Pipfile, Pipfile.lock
- Testing; tox.ini

Along with several more.

Signed-off-by: Jason Joyce <fuzzball81@gmail.com>

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+pip==9.0.1
```
Comment (jguiditta) [inline requirements_dev.txt]: Just to make sure, do you have Pipfile and this requirements_dev.txt on purpose?  I am guessing this is for rtd maybe?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+setup(
+    author="Jason Joyce",
+    author_email='fuzzball81@gmail.com',
+    classifiers=[
+        'Development Status :: 2 - Pre-Alpha',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: MIT License',
+        'Natural Language :: English',
+        "Programming Language :: Python :: 2",
```
Comment (jguiditta) [inline setup.py]: Probably don't want python 2 listed here?

**Review by jguiditta — COMMENTED** [review]
Couple comments, looks good otherwise.

File: setup.py (hunk 2 above)
Comment (lhh) [reply setup.py]: Yeah; should be python :: 3

**Review by jguiditta — APPROVED** [review]

**Review by lhh — APPROVED** [review]

**Pull request merged**